### PR TITLE
feat!: added Menu component family using Reach UI

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
### PR Checklist

- [x] Related to designs: https://www.figma.com/file/7PNKwpTpXjRrPjcCRSu215/Global-Nav?node-id=932%3A3085
- [x] Related to JIRA ticket: [WEB-1167](https://codecademy.atlassian.net/browse/WEB-1167)
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### Description

Directly uses a large portion of Reach UI's [Menu Button](https://reach.tech/menu-button/#reach-skip-nav) family, with the addition of a `MenuButtonToggle` for the chevron icon expando logic. 
[Slack thread on avoiding Popover directly](https://codecademy.slack.com/archives/C047ESP1T/p1605631142239000)

Renames the old `SideMenu` component's `MenuItem` to `SideMenuItem` to distinguish it from the `MenuItem` here.

See also https://reach.tech/menu-button for now the menu button normally behaves. I was surprised that you can't use tabbing inside it when open, just up/down arrow keys...

Toggling the menu popover

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.3.3-alpha.52b66c.0<br/>@codecademy/gamut@17.4.0-alpha.52b66c.0<br/>@codecademy/styleguide@18.6.0-alpha.52b66c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fb435f968b297012f37812e--gamut-preview.netlify.app>https://5fb435f968b297012f37812e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fb435f968b297012f37812e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.5.2-alpha.505929.0<br/>@codecademy/gamut@17.4.0-alpha.505929.0<br/>@codecademy/styleguide@18.9.0-alpha.505929.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fb6dd1bfdba271aa16948fd--gamut-preview.netlify.app>https://5fb6dd1bfdba271aa16948fd--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fb6dd1bfdba271aa16948fd>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.5.2-alpha.71798b.0<br/>@codecademy/gamut@17.4.0-alpha.71798b.0<br/>@codecademy/styleguide@18.9.0-alpha.71798b.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.5.2-alpha.e2f6ab.0<br/>@codecademy/gamut@17.4.0-alpha.e2f6ab.0<br/>@codecademy/styleguide@18.9.0-alpha.e2f6ab.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fb6e042c8cb291dfaae98f0--gamut-preview.netlify.app>https://5fb6e042c8cb291dfaae98f0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fb6e042c8cb291dfaae98f0>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fb6e07e9e2208270192fb8f--gamut-preview.netlify.app>https://5fb6e07e9e2208270192fb8f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fb6e07e9e2208270192fb8f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.5.2-alpha.0ff04e.0<br/>@codecademy/gamut@17.4.0-alpha.0ff04e.0<br/>@codecademy/styleguide@18.9.0-alpha.0ff04e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fb6e0fd9d33a31bfc0f8249--gamut-preview.netlify.app>https://5fb6e0fd9d33a31bfc0f8249--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fb6e0fd9d33a31bfc0f8249>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.5.2-alpha.65ca87.0<br/>@codecademy/gamut@17.4.0-alpha.65ca87.0<br/>@codecademy/styleguide@18.9.0-alpha.65ca87.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fb6e3669d33a31a4e0f8338--gamut-preview.netlify.app>https://5fb6e3669d33a31a4e0f8338--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fb6e3669d33a31a4e0f8338>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.5.2-alpha.9d3a4f.0<br/>@codecademy/gamut@17.4.0-alpha.9d3a4f.0<br/>@codecademy/styleguide@18.9.0-alpha.9d3a4f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fb6e6396f61f71e6752c27e--gamut-preview.netlify.app>https://5fb6e6396f61f71e6752c27e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fb6e6396f61f71e6752c27e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.5.2-alpha.6889d7.0<br/>@codecademy/gamut@17.4.0-alpha.6889d7.0<br/>@codecademy/styleguide@18.9.0-alpha.6889d7.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fb6ed550f56dd3411821123--gamut-preview.netlify.app>https://5fb6ed550f56dd3411821123--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fb6ed550f56dd3411821123>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.5.2-alpha.759e53.0<br/>@codecademy/gamut@17.4.0-alpha.759e53.0<br/>@codecademy/styleguide@18.9.0-alpha.759e53.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fb6eda42c08872a5f75fd62--gamut-preview.netlify.app>https://5fb6eda42c08872a5f75fd62--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fb6eda42c08872a5f75fd62>Deploy Logs</a></p>

File: packages/gamut/src/Menu/override-reach-ui.css [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/*
+https://reach.tech/styling/
+Their docs are pretty clear that it's normally better to use their styles as a base.
+...as we use more Reach components, we should investigate switching over.
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Menu/override-reach-ui.css:4]: If you want to see why I disabled this, remove the CSS variable locally and observe the dark blue backgrounds. I didn't see any real changes other than colors so unless I'm missing something, I think this is just an easier & quicker implementation for now. Thoughts?

Comment (JoshuaKGoldberg) [reply packages/gamut/src/Menu/override-reach-ui.css:4]: You know what, I'm just gonna go ahead and use their thing...

Comment (JoshuaKGoldberg) [reply packages/gamut/src/Menu/override-reach-ui.css]: Yeah it was just the one background color. I was being silly 😄 .

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.5.2-alpha.fed6b8.0<br/>@codecademy/gamut@17.4.0-alpha.fed6b8.0<br/>@codecademy/styleguide@18.9.0-alpha.fed6b8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fb6f25edf5bdf33ddd1e876--gamut-preview.netlify.app>https://5fb6f25edf5bdf33ddd1e876--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fb6f25edf5bdf33ddd1e876>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.5.2-alpha.70b258.0<br/>@codecademy/gamut@17.4.0-alpha.70b258.0<br/>@codecademy/styleguide@18.9.0-alpha.70b258.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fb6feb92b5ca10c6c221d25--gamut-preview.netlify.app>https://5fb6feb92b5ca10c6c221d25--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fb6feb92b5ca10c6c221d25>Deploy Logs</a></p>

File: packages/gamut/src/Menu/styles.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { colors, spacing, swatches } from '@codecademy/gamut-styles';
+import { css } from '@emotion/react';
+
+export const listItemStyles = css`
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Menu/styles.ts:4]: These aren't _quite_ exactly what we need, but I'll tweak them tomorrow / Monday if everything else goes well...

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.5.2-alpha.8c99e6.0<br/>@codecademy/gamut@17.4.0-alpha.8c99e6.0<br/>@codecademy/styleguide@18.9.0-alpha.8c99e6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fb7020c07e2640f46ff5634--gamut-preview.netlify.app>https://5fb7020c07e2640f46ff5634--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fb7020c07e2640f46ff5634>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.5.2-alpha.a48252.0<br/>@codecademy/gamut@17.4.0-alpha.a48252.0<br/>@codecademy/styleguide@18.9.0-alpha.a48252.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fbaabf0c2bc9568ba81c281--gamut-preview.netlify.app>https://5fbaabf0c2bc9568ba81c281--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fbaabf0c2bc9568ba81c281>Deploy Logs</a></p>

File: packages/gamut/src/Menu/MenuLink.tsx [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import styled from '@emotion/styled';
+import { MenuLink as ReachMenuLink } from '@reach/menu-button';
+
+import { listItemStyles } from './styles';
+
+export const MenuLink = styled(ReachMenuLink)`
+  ${listItemStyles}
```
Comment (sooinc) [inline packages/gamut/src/Menu/MenuLink.tsx:7]: ooh didn't know we can style this way! thx for teaching me

File: packages/gamut/src/Menu/MenuButtonToggle.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  isExpanded?: boolean;
+};
+
+const StyledIcon = styled(ArrowChevronDownFilledIcon)<{ isExpanded?: boolean }>`
+  margin-left: ${pxRem(5)};
+
+  ${({ isExpanded }) =>
+    isExpanded &&
+    `
+    margin-bottom: 3px;
```
Comment (jcenglish) [inline packages/gamut/src/Menu/MenuButtonToggle.tsx:19]: pxRem here?

Comment (JoshuaKGoldberg) [reply packages/gamut/src/Menu/MenuButtonToggle.tsx]: Aha, yes, I actually just ended up removing this (it's not there on prod).

File: packages/gamut/src/Menu/styles.ts (hunk 2 above)
Comment (JoshuaKGoldberg) [reply packages/gamut/src/Menu/styles.ts:4]: Tweaked in the latest version 😄

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.5.2-alpha.36e7f1.0<br/>@codecademy/gamut@17.4.0-alpha.36e7f1.0<br/>@codecademy/styleguide@18.9.0-alpha.36e7f1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fbbfce2f886ab1978efba54--gamut-preview.netlify.app>https://5fbbfce2f886ab1978efba54--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fbbfce2f886ab1978efba54>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.5.2-alpha.882d27.0<br/>@codecademy/gamut@17.4.0-alpha.882d27.0<br/>@codecademy/styleguide@18.9.0-alpha.882d27.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fbbfe938f7e981956af5e08--gamut-preview.netlify.app>https://5fbbfe938f7e981956af5e08--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fbbfe938f7e981956af5e08>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.5.2-alpha.1d7420.0<br/>@codecademy/gamut@17.4.0-alpha.1d7420.0<br/>@codecademy/styleguide@18.9.0-alpha.1d7420.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fbc0399efbe1d1a1f01fcac--gamut-preview.netlify.app>https://5fbc0399efbe1d1a1f01fcac--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fbc0399efbe1d1a1f01fcac>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.7.3-alpha.ae76d4.0<br/>@codecademy/gamut@17.5.0-alpha.ae76d4.0<br/>@codecademy/styleguide@18.11.0-alpha.ae76d4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fbc0563f886ab1efbefbe99--gamut-preview.netlify.app>https://5fbc0563f886ab1efbefbe99--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fbc0563f886ab1efbefbe99>Deploy Logs</a></p>

**Comment by jcenglish** [comment]
I think it'd be helpful for the submenus' `a` tags to have `aria-expanded='true'` and `aria-expanded='false'` attributes when they're open and closed. That way, screenreader users know for sure it's a button that reveals more navigation options and not some other kind of button with unpredictable behavior. That's how it is in [this W3 example](https://www.w3.org/TR/wai-aria-practices/examples/menubar/menubar-1/menubar-1.html).

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.7.3-alpha.38b19e.0<br/>@codecademy/gamut@17.5.0-alpha.38b19e.0<br/>@codecademy/styleguide@18.11.0-alpha.38b19e.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.7.3-alpha.4f7aef.0<br/>@codecademy/gamut@17.5.0-alpha.4f7aef.0<br/>@codecademy/styleguide@18.11.0-alpha.4f7aef.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fbc17839914bd3797201e14--gamut-preview.netlify.app>https://5fbc17839914bd3797201e14--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fbc17839914bd3797201e14>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fbc17a85e6e99351b19301d--gamut-preview.netlify.app>https://5fbc17a85e6e99351b19301d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fbc17a85e6e99351b19301d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.7.3-alpha.f63f3e.0<br/>@codecademy/gamut@17.5.0-alpha.f63f3e.0<br/>@codecademy/styleguide@18.11.0-alpha.f63f3e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fbc254060b25f4bb21cb4b3--gamut-preview.netlify.app>https://5fbc254060b25f4bb21cb4b3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fbc254060b25f4bb21cb4b3>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.7.3-alpha.394af1.0<br/>@codecademy/gamut@17.5.0-alpha.394af1.0<br/>@codecademy/styleguide@18.11.0-alpha.394af1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fbce35564a2610090ce5c2e--gamut-preview.netlify.app>https://5fbce35564a2610090ce5c2e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fbce35564a2610090ce5c2e>Deploy Logs</a></p>

File: packages/gamut/src/Menu/MenuButton.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  border-radius: 2px;
+  cursor: pointer;
+  padding: ${spacing[4]} ${spacing[8]};
+  transition: box-shadow 0.1s ease-in-out;
+
+  ${({ isExpanded }) => isExpanded && `font-weight: bold;`}
+
+  &[data-focus-visible-added] {
+    box-shadow: 0 0 0 2px ${colors.black};
+  }
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Menu/MenuButton.tsx:18]: Pending: https://github.com/codecademy-engineering/Codecademy/pull/20352#discussion_r528977774

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.7.3-alpha.0be953.0<br/>@codecademy/gamut@17.5.0-alpha.0be953.0<br/>@codecademy/styleguide@18.11.0-alpha.0be953.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fbe51349261666cdb8de86c--gamut-preview.netlify.app>https://5fbe51349261666cdb8de86c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fbe51349261666cdb8de86c>Deploy Logs</a></p>

**Comment by JoshuaKGoldberg** [comment]
Closing to keep my PR queue small.

**Pull request closed without merging**